lib/node_modules/@stdlib/utils/parse-dsv/lib/main.js

```diff
@@ -0,0 +1,369 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2018 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var EventEmitter = require( 'events' ).EventEmitter;
+var setReadOnly = require( '@stdlib/utils/define-nonenumerable-read-only-property' );
+var instanceOf = require( '@stdlib/assert/instance-of' );
+var isBuffer = require( '@stdlib/assert/is-buffer' );
+var inherit = require( '@stdlib/utils/inherit' );
+
+
+// VARIABLES //
+
+var CR = '\r';
+var LF = '\n';
+var CRLF = '\r\n';
+
+var TOKEN_ENUM = {
+	'DELIM': 0,
+	'QUOTE': 1,
+	'CR': 2,
+	'LF': 3,
+	'CRLF': 4,
+	'TEXTDATA': 5,
+	'ENDOFSTREAM': 6
+};
+
+var STATE_ENUM = {
+	'INITIAL': 0,
+	'START': 1,
+	'RECORD': 2,
+	'ESCAPED': 3,
+	'NONESCAPED': 4,
+	'TWOQUOTE': 5,
+	'END': 6
+};
+
+
+// MAIN //
+
+/* eslint-disable no-invalid-this */
+
+/**
+* Returns an a function which incrementally a delimiter separated values resource.
+* @param {Object} [options] - something
+* @returns {Function} accumulator function
+*
+* @example
+* var string2buffer = require( '@stdlib/buffer/from-string' );
+* var parse_dsv = new parseDsvStream();
+*
+* parse_dsv.incrparse( string2buffer( '1,2,3' ) );
+*
+* parse_dsv.incrparse( string2buffer( '\n' ) );
+*
+* parse_dsv.incrparse( string2buffer('4, 5') );
+*
+* parse_dsv.incrparse( string2buffer( ', 6' ) );
+*/
+function parseDsvStream( options = {} ) { // Fix class name later
+	if ( !instanceOf( this, parseDsvStream ) ) {
+		return new parseDsvStream( options );
+	}
+	this.QUOTECHAR = options.QUOTECHAR || '"';
```

Comment:
You'll need to do `EventEmitter.call( this )` in here to ensure that parent instance properties are properly bound.